### PR TITLE
Add additional expression tests

### PR DESCRIPTION
**Pull request by ozevren**
Wrote these tests to quickly verify the evaluator behavior outlined in https://github.com/istio/istio/issues/3734.

### CODE REVIEW

File: mixer/pkg/il/testing/tests.go [hunk 1]
```diff
@@ -305,6 +305,50 @@ end`,
 		R:    true,
 		conf: exprEvalAttrs,
 	},
+	// use request.header (without "s" at the end), as it is the one in the original attributeset.
```
Comment (douglas-reid) [inline mixer/pkg/il/testing/tests.go]: should we update the testing set to match the actual vocab (`request.headers`) so as not to sow confusion?

Comment (ozevren) [reply mixer/pkg/il/testing/tests.go]: Added 0.6 attribute set as a separate set.

I'd rather not pollute the previous tests that I copied over from previous expression tests.

**Pull request merged**